### DISCUSSION TITLE
[Feature Request] Automatic Updates

### DISCUSSION THREAD

**Discussion by holocronology** [louislam/dockge#32] [Feature Request]
This is perhaps a down the line feature...

The goal here would be to match the features of Watchtower. Add the following:

- Selector to enable automatic updates for a specific container
- cron settings for automatic updates
- separate log or indicator on container of "last updated on: DATE/TIME"
- notifications

OR, some way to integrate with Watchtower ...

**Comment by PsychotherapistSam** [comment]
Personally I don't like automatic updates a lot, but having a label in the list of containers showing it's out of date would be neat, which might fit in with this request.

**Comment by RichyHBM** [comment]
Maybe this could be achieved using an integration with https://github.com/fmartinou/whats-up-docker, that way you could manage it all via labels etc, and either define a periodic time period to check or add a new trigger to have wud automatically notify dockge

**Comment by collse** [comment]
while I am not a fan of automatic updates either, I would think what would help is an "update available" notification, where on a scheduled basis existence of a newer image is being checked and then a notification is displayed on the stack

**Reply by nathan-dan** [reply]
This functionality is currently in Portainer & it would be great to see here. I’m also not a fan of automatic updates so I have my stacks monitored by [Diun ](https://github.com/crazy-max/diun). While notifications via Diun let me know that there are updates available, it would be good to see which containers need updating.

**Reply by wpbezemer** [reply]
Additionally a publication on MQTT as notification would also be really helpful for the IOT guys of us. 

And thirdly, an API to update individual or all images. 

Currently I’m using watchtower + what’s up docker + portainer.. would like to have one for all.

**Comment by nnmalex** [comment]
There's a great project that implements a bash script that allows checking for updates - [dockcheck.](https://github.com/mag37/dockcheck) Perhaps it can be used for automated checks in Dockge?

**Comment by Tavisco** [comment]
I'm not a fan of automatics updates either, but notifications for updates would be AWESOME! Dockge could probably leverage the notification mechanism that already exists on Kuma. And of course, label the containers as "Update avaliable".

**Comment by qroac** [comment]
I just came to create a post to request an update indicator feature.

Would be great to have an update-icon in the stack list if a container in the stack is outdated and more info on the detail page.

I'd further love to have available updates found on two different ways.
- new image for the specific image tag in use, e.g. `alpine:latest` or `some/app:stable`
- newer semantic versioned tag on the repository, e.g.
  - `image:3` -> `image:4`
  - `image:release-2.1` -> `image:release-2.2` or `image:release-3.0`
  - down to 3 or 4 levels to allow fine controlled updates

**Comment by opticsthewebsite** [comment]
This would be a great feature and would make Dockge nearly perfect for me.

**Comment by jcz1** [comment]
Also "Update all" button for update of all stacks would be great.

**Reply by chherman0** [reply]
I would love an update all button

**Comment by gizmocuz** [comment]
Are there any plans to support this? If someone is willing to make a PR would you consider accepting this? (Only the Update notification/display), I encourage users to update their instances one-by-one, and also check if it has succeeded.
An UpdateAll I personally would  never ever press

**Comment by GamingDaveUk** [comment]
new to docker, and dockage has been very handy, was looking to see how i could automate updates.... found this unanswered feature request. So plus one for updating on a schedule.

**Comment by finder39** [comment]
Coming back to this thread since most of what's been requested here now exists in my fork:

- **Per-stack auto-update toggle** with cron scheduler — @GamingDaveUk, OP
- **Update available indicators** in the stack list — @PsychotherapistSam, @collse, @qroac, @Tavisco
- **Update All button** on the dashboard — @jcz1, @chherman0
- **REST API** for updating individual or all stacks — @wpbezemer
- **Update history log** with timestamps and output — OP's "last updated on" request
- **Home Assistant integration** with sensors and buttons — @wpbezemer's MQTT/IoT request (not MQTT, but achieves the same thing via REST)

Image update detection uses skopeo to compare local vs remote digests — no Watchtower or external tools needed.

Transparency note: this was entirely vibecoded with Claude Code. I'm an iOS developer, not a backend/Docker guy, so this was a fun side project. It works well for my setup (4 servers, ~30 stacks) but YMMV.

Full details: [discussion post](https://github.com/louislam/dockge discussions/844#discussioncomment-12733498) | [finder39/dockge](https://github.com/finder39/dockge) | `ghcr.io/finder39/dockge:release` | [HA integration](https://github.com/finder39/ha-dockge)

**Reply by wpbezemer** [reply]
That sounds really awesome, great job!
I will test this probably this week. Thanks so much!
